src/allmydata/crypto/ed25519.py

```diff
@@ -0,0 +1,225 @@
+'''
+Ed25519 keys and helpers.
+
+Key Formatting
+--------------
+
+- in base32, keys are 52 chars long (both signing and verifying keys)
+- in base62, keys is 43 chars long
+- in base64, keys is 43 chars long
+
+We can't use base64 because we want to reserve punctuation and preserve
+cut-and-pasteability. The base62 encoding is shorter than the base32 form,
+but the minor usability improvement is not worth the documentation and
+specification confusion of using a non-standard encoding. So we stick with
+base32.
+'''
+
+import six
+
+from cryptography.exceptions import (
+    InvalidSignature,
+)
+from cryptography.hazmat.primitives.asymmetric.ed25519 import (
+    Ed25519PrivateKey,
+    Ed25519PublicKey,
+)
+from cryptography.hazmat.primitives.serialization import (
+    Encoding,
+    PrivateFormat,
+    NoEncryption,
+    PublicFormat,
+)
+
+from allmydata.crypto import (
+    remove_prefix,
+    BadSignature,
+)
+from allmydata.util.base32 import (
+    a2b,
+    b2a,
+)
+
+PRIVATE_KEY_PREFIX = 'priv-v0-'
+PUBLIC_KEY_PREFIX = 'pub-v0-'
+
+
+def create_signing_keypair():
+    """
+    Creates a new ed25519 keypair.
+
+    :returns: 2-tuple of (private_key, public_key)
```

Comment:
Possibly another place where a marker interface would help with clarity

Reply:
I tried to run this code and I encountered a great deal of difficulty getting a version of openssl with which I could actually use cryptography's ed25519 APIs. :/

I can't tell what version of openssl introduced this feature.  cryptography thinks 1.1.1b is too old.  1.1.1.c was only released in May.  The openssl changelog does nothing to tell me which version I might need.  Yet on the CLI 1.1.1b seems like it _can_ generate ed25519 keys...

Reply:
> reaperhulk | exarkun: they have a bug in OpenSSL < 1.1.1c that allows signature malleability in ed25519

:(

Reply:
Hrmmm, okay so we also happen to depend on OpenSSL >= 1.1.1c then..? Can we inform `pip` of this?

Reply:
fwiw I did finally manage to upgrade to 1.1.1c (pip's wheel cache defeated me for a while, keeping the binary wheel built against older openssl around and constantly re-installing it instead of building a new one) and this let me run this ed25519 code.  And I verified that the old and new signing code produces the same result given the same key and data.

> Hrmmm, okay so we also happen to depend on OpenSSL >= 1.1.1c then..?

I think yes.

> Can we inform pip of this?

I think ... no? :/

Reply:
I added notes to the docs